src/Java.Interop/Java.Interop/JniRuntime.JniMarshalMemberBuilder.cs

```diff
@@ -31,6 +31,7 @@ partial class CreationOptions {
 #if NET
 		[DynamicDependency (DynamicallyAccessedMemberTypes.PublicParameterlessConstructor, "Java.Interop.MarshalMemberBuilder", "Java.Interop.Export")]
 		[UnconditionalSuppressMessage ("Trimming", "IL2026", Justification = "DynamicDependency should preserve the constructor.")]
+		[UnconditionalSuppressMessage ("Trimming", "IL2072", Justification = "DynamicDependency should preserve the constructor.")]
 		[UnconditionalSuppressMessage ("Trimming", "IL2035", Justification = "Java.Interop.Export.dll is not always present.")]
 #endif
 		partial void SetMarshalMemberBuilder (CreationOptions options)
```

Comment:
Is there a reason this can't use `Type.GetType("Java.Interop.MarshalMemberBuilder, Java.Interop.Export")`.
That is something both trimmer and AOT understand and will handle without warnings and correctly preserve the .ctor in the call to `Activator.CreateInstance`.

Reply:
This is kind of an odd codepath. We have a `Java.Interop.Export.dll` in this repo, but we don't actually ship it in the Android workload. Some of the projects in here could be used for Java interop on desktop, etc., but there isn't a real product using them.

Let me check if it's worth refactoring this yet.

Reply:
As a scenario this is not new - there are many places even in the core BCL where something similar happens. In fact there are a couple of places even in CoreLib where we use this to break circular references between assemblies. Meaning it's a pattern which trimming/AOT has to support well.